layered-architectures-module-2/Catalog.Application/Products/Commands/DeleteProduct/DeleteProductCommand.cs

```diff
@@ -0,0 +1,40 @@
+﻿using Catalog.Application.Common.Exceptions;
+using Catalog.Application.Common.Interfaces;
+using Catalog.Domain.Entities;
+using MediatR;
+using Microsoft.EntityFrameworkCore;
+
+namespace Catalog.Application.Products.Commands.DeleteProduct;
+
+public class DeleteProductCommand : IRequest
+{
+    public int Id { get; set; }
+}
+
+public class DeleteProductCommandHandler : IRequestHandler<DeleteProductCommand>
+{
+    private readonly IApplicationDbContext _context;
+
+    public DeleteProductCommandHandler(IApplicationDbContext context)
+    {
+        _context = context;
+    }
+
+    public async Task<Unit> Handle(DeleteProductCommand request, CancellationToken cancellationToken)
+    {
+        var entity = await _context.Products
+            .Where(l => l.Id == request.Id)
+            .SingleOrDefaultAsync(cancellationToken);
```

Comment:
do you know the diff between SingleOrDefault vs FirstOrDefaultAsync? 
do you know FindAsync?

Reply:
You're right, I considered to use FindAsync, suit better here